### PR TITLE
configuration: Add set-default-browser.

### PR DESCRIPTION
**Pull request by Ambrevar**
# Description

Add a low-level function to set the default browser.

Preparation for #2351.

# Discussion

This leverages the py-configparser library.  Does the job.
Anything else you would recommend otherwise?

# Checklist:
Everything in this checklist is required for each PR.  Please do not approve a PR that does not have all of these items.  

- [x] I have pulled from master before submitting this PR
- [x] My code follows the style guidelines for Common Lisp code
  - See [Norvig & Pitman's Tutorial on Good Lisp Programming Style (PDF)](https://www.cs.umd.edu/~nau/cmsc421/norvig-lisp-style.pdf) and [Google Common Lisp Style Guide](https://google.github.io/styleguide/lispguide.xml).
- [x] I have performed a self-review of my own code
- [ ] My code has been reviewed by at least one peer (the peer review to approve a PR counts.  The reviewer must download and test the code)
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

### CODE REVIEW

File: source/configuration.lisp [hunk 1]
```diff
@@ -346,6 +346,35 @@ Call this function from your initialization file to re-enable the default ASDF r
...
+                                     (uiop:xdg-data-home "applications/mimeapps.list"))))
+  "Return the modified MIME apps list.
+Return the persisted file as second value."
+  (let* ((target (or (first (sera:filter #'uiop:file-exists-p targets))
+                     (first targets)))
+         (config (py-configparser:read-files (py-configparser:make-config)
+                                             (list target)))
+         (desktop-file (uiop:strcat name ".desktop")))
+    (dolist (section '("Added Associations" "Default Applications"))
+      (dolist (key '("text/html"
```
Comment (aartaka) [inline source/configuration.lisp:361]: plus `text/gemini`!

**Comment by aartaka** [comment]
Maybe rather `set-as-default-browser`? The name is confusing, I thought it's about setting some meta-class or something...

File: source/configuration.lisp (hunk 1 above)
Comment (aartaka) [reply source/configuration.lisp:361]: No, Gopher has no dedicated text/media MIMEs. I doubt it has any concept of MIMEs at all xD

**Comment by Ambrevar** [comment]
Should be ready to merge.

**Comment by Ambrevar** [comment]
Actually, nope!  This is XDG specific and won't work on macOS or Windows.  We need to add a feature macro here.  Which one should it be?

**Comment by jmercouris** [comment]
> Actually, nope! This is XDG specific and won't work on macOS or Windows. We need to add a feature macro here. Which one should it be?



> Actually, nope! This is XDG specific and won't work on macOS or Windows. We need to add a feature macro here. Which one should it be?

Would it not be #+linux #-darwin or something like that?

**Comment by Ambrevar** [comment]
I think we need a flag that encompasses all distros that run Xorg / Waylang, so the *BSD as well, Solaris, etc. but not macOS.

Maybe `#+(and unix (not darwin)) `?

**Comment by aartaka** [comment]
> Maybe `#+(and unix (not darwin)) `?

Yes, should work.

**Comment by jmercouris** [comment]
Yes, this would work.

**Pull request closed without merging**